test/unit/org/apache/cassandra/db/guardrails/GuardrailPartitionKeysInSelectTest.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+
+public class GuardrailPartitionKeysInSelectTest extends ThresholdTester
+{
+    private static final int PARTITION_KEYS_SELECT_WARN_THRESHOLD = 3;
+    private static final int PARTITION_KEYS_SELECT_ABORT_THRESHOLD = 5;
+
+    public GuardrailPartitionKeysInSelectTest()
+    {
+        super(PARTITION_KEYS_SELECT_WARN_THRESHOLD,
+              PARTITION_KEYS_SELECT_ABORT_THRESHOLD,
+              DatabaseDescriptor.getGuardrailsConfig().getPartitionKeysInSelect(),
+              Guardrails::setPartitionKeysInSelectThreshold,
+              Guardrails::getPartitionKeysInSelectWarnThreshold,
+              Guardrails::getPartitionKeysInSelectAbortThreshold);
+    }
+
+    @Before
+    public void setupTest()
+    {
+        createKeyspace("CREATE KEYSPACE IF NOT EXISTS partition_keys_in_select_test_ks " +
+                       "with replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 } " +
+                       "and durable_writes = false");
+        createTable("CREATE TABLE IF NOT EXISTS partition_keys_in_select_test_ks.partition_keys_in_select_test " +
+                    "(k INT, c INT, v TEXT, PRIMARY KEY(k, c))");
+    }
+
+    @Test
+    public void testSelectStatementAgainstInClausePartitionKeys() throws Throwable
+    {
+        assertValid("SELECT k, c, v FROM partition_keys_in_select_test_ks.partition_keys_in_select_test " +
+                    "WHERE k=10");
+
+        assertValid("SELECT k, c, v FROM partition_keys_in_select_test_ks.partition_keys_in_select_test " +
+                    "WHERE k IN (2, 3)");
+
+        assertValid("SELECT k, c, v FROM partition_keys_in_select_test_ks.partition_keys_in_select_test " +
+                    "WHERE k = 2 and c IN (2, 3, 4, 5, 6, 7)");
+
+        assertWarns("Query with partition keys in IN clause on table partition_keys_in_select_test, " +
+                    "with number of partition keys 4 exceeds warning threshold of 3.",
+                    "SELECT k, c, v FROM partition_keys_in_select_test_ks.partition_keys_in_select_test " +
+                        "WHERE k IN (2, 3, 4, 5)");
+
+        assertAborts("Aborting query with partition keys in IN clause on table partition_keys_in_select_test, " +
+                     "number of partition keys 6 exceeds abort threshold of 5." ,
+                     "SELECT k, c, v FROM partition_keys_in_select_test_ks.partition_keys_in_select_test " +
+                     "WHERE k IN (2, 3, 4, 5, 6, 7)"
+                     );
```

Comment:
Instead of creating a keyspace we can use the default keyspace for tests and autogenerated table name, similarly to what's done for example in [`GuardrailPageSizeTest`](https://github.com/apache/cassandra/blob/trunk/test/unit/org/apache/cassandra/db/guardrails/GuardrailPageSizeTest.java). It's not a big difference but I think it makes the test a bit more concise and, more importantly, it makes sure that the table is removed after finishing the test:
```suggestion
        createTable("CREATE TABLE %s (k INT, c INT, v TEXT, PRIMARY KEY(k, c))");
    }

    @Test
    public void testSelectStatementAgainstInClausePartitionKeys() throws Throwable
    {
        assertValid("SELECT k, c, v FROM %s WHERE k=10");

        assertValid("SELECT k, c, v FROM %s WHERE k IN (2, 3)");

        assertValid("SELECT k, c, v FROM %s WHERE k = 2 and c IN (2, 3, 4, 5, 6, 7)");

        assertWarns(format("Query with partition keys in IN clause on table %s, " +
                           "with number of partition keys 4 exceeds warning threshold of 3.",
                           currentTable()),
                    "SELECT k, c, v FROM %s WHERE k IN (2, 3, 4, 5)");

        assertAborts(format("Aborting query with partition keys in IN clause on table %s, " +
                            "number of partition keys 6 exceeds abort threshold of 5.",
                            currentTable()),
                     "SELECT k, c, v FROM %s WHERE k IN (2, 3, 4, 5, 6, 7)");
```

Reply:
👍